Engine/game/savegame.cpp

```diff
@@ -0,0 +1,596 @@
+            if (r_data.RoomBkgScene[i])
+            {
+                delete thisroom.ebscene[i];
+                thisroom.ebscene[i] = r_data.RoomBkgScene[i];
+            }
+        }
+
+        in_new_room=3;  // don't run "enters screen" events
+        // now that room has loaded, copy saved light levels in
+        memcpy(thisroom.regionLightLevel, r_data.RoomLightLevels, sizeof(short) * MAX_REGIONS);
+        memcpy(thisroom.regionTintLevel, r_data.RoomTintLevels, sizeof(int) * MAX_REGIONS);
+        generate_light_table();
+
+        memcpy(thisroom.walk_area_zoom, r_data.RoomZoomLevels1, sizeof(short) * (MAX_WALK_AREAS + 1));
+        memcpy(thisroom.walk_area_zoom2, r_data.RoomZoomLevels2, sizeof(short) * (MAX_WALK_AREAS + 1));
+
+        on_background_frame_change();
+    }
+
+    gui_disabled_style = convert_gui_disabled_style(game.options[OPT_DISABLEOFF]);
+
+    // restore the queue now that the music is playing
+    play.music_queue_size = queuedMusicSize;
+
+    if (play.digital_master_volume >= 0)
+        System_SetVolume(play.digital_master_volume);
+
+    // Run audio clips on channels
+    // these two crossfading parameters have to be temporarily reset
+    const int cf_in_chan = play.crossfading_in_channel;
+    const int cf_out_chan = play.crossfading_out_channel;
+    play.crossfading_in_channel = 0;
+    play.crossfading_out_channel = 0;
+    for (int i = 0; i <= MAX_SOUND_CHANNELS; ++i)
+    {
+        const RestoredData::ChannelInfo &chan_info = r_data.AudioChans[i];
+        if (chan_info.ClipID < 0)
+            continue;
+        if (chan_info.ClipID >= game.audioClipCount)
+        {
+            Out::FPrint("Restore game error: invalid audio clip index: %d (clip count: %d)", chan_info.ClipID, game.audioClipCount);
+            return kSvgErr_GameObjectInitFailed;
+        }
+        play_audio_clip_on_channel(i, &game.audioClips[chan_info.ClipID],
+            chan_info.Priority, chan_info.Repeat, chan_info.Pos);
+        if (channels[i] != NULL)
+        {
+            channels[i]->set_volume_alternate(chan_info.VolAsPercent, chan_info.Vol);
+            channels[i]->set_speed(chan_info.Speed);
+            channels[i]->set_panning(chan_info.Pan);
+            channels[i]->panningAsPercentage = chan_info.PanAsPercent;
+        }
+    }
+    if ((cf_in_chan > 0) && (channels[cf_in_chan] != NULL))
+        play.crossfading_in_channel = cf_in_chan;
+    if ((cf_out_chan > 0) && (channels[cf_out_chan] != NULL))
+        play.crossfading_out_channel = cf_out_chan;
+
+    // If there were synced audio tracks, the time taken to load in the
+    // different channels will have thrown them out of sync, so re-time it
+    for (int i = 0; i <= MAX_SOUND_CHANNELS; ++i)
+    {
+        int pos = r_data.AudioChans[i].Pos;
+        if ((pos > 0) && (channels[i] != NULL) && (channels[i]->done == 0))
+        {
+            channels[i]->seek(pos);
+        }
+    }
+
+    for (int i = 1; i < MAX_SOUND_CHANNELS; ++i)
```

Comment:
I would, but what do I write in these comments?
This code needs investigation, because I do not remember (and possibly never knew) why does it look this way.
I honestly cannot even say if that is not a bug. I can only check that it was originally same way.